README.md

```diff
@@ -1 +1,186 @@
+
+[npm-url]:https://npmjs.org/package/@moxy/react-wait-for-react
+[downloads-image]:https://img.shields.io/npm/dm/@moxy/react-wait-for-react.svg
+[npm-image]:https://img.shields.io/npm/v/@moxy/react-wait-for-react.svg
+[build-status-url]:https://github.com/moxystudio/next-with-moxy/actions
+[build-status-image]:https://img.shields.io/github/workflow/status/moxystudio/next-with-moxy/Node%20CI/master
+[codecov-url]:https://codecov.io/gh/moxystudio/react-wait-for-react
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/react-wait-for-react/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/react-wait-for-react
+[david-dm-image]:https://img.shields.io/david/moxystudio/react-wait-for-react.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/react-wait-for-react?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/react-wait-for-react.svg
+
+Easily render a loader while your React app is loading, optionally waiting for a promise as well.
+
+## Installation
+
+```sh
+$ npm install @moxy/react-wait-for-react
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+Certain apps or pages have impactful experiences. These experiences can make the total bundle size larger as they pack possibly large dependencies and media assets, such as 3D objects and audio files.
+
+It's then often normal to preload all the required files for an uninterrupted experience. `@moxy/wait-for-react` is a library that makes it easy to display a loader before your static or server-side rendered app becomes interactive, and optionally until all the required files are loaded (via a promise).
+
+<img src="https://developers.google.com/web/fundamentals/performance/images/perf-metrics-load-timeline.png" alt="Performance metrics timeline" width="700" />
+
+⚠️ You should still render the app or page contents "below" the loader, to keep your website SEO friendly.
+
+## Demo
+
+You may see a simple demo of `react-wait-for-react` in [https://moxystudio.github.io/react-wait-for-react](https://moxystudio.github.io/react-wait-for-react/).
+
+## Usage
+
+Using `<WaitForReact>` to render a progress bar while your page assets are being loaded:
+
+```js
+import React from 'react';
+import classNames from 'classnames';
+import WaitForReact from '@moxy/react-wait-for-react';
+import styles from './MyPage.module.css';
+
+const preloadAssets = async () => {
+    // Preload files, like a mp3, 3d objects, etc..
+};
+
+const MyPage = () => {
+    const promise = useMemo(() => preloadAssets(), []);
+    const applyProgressBeforeInteractive = useCallback((elements, progress) =>
```

Comment:
Shouldn't the examples show that applyProgressBeforeInteractive should be a string?

Reply:
Yep, fixed.